neutrino-custom/base.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+const path = require('path');
+const webpack = require('webpack');
+const reactPreset = require('neutrino-preset-react');
+const HtmlPlugin = require('html-webpack-plugin');
+
+const CWD = process.cwd();
+const SRC = path.join(CWD, 'src'); // neutrino's default source directory
+const UI = path.join(CWD, 'ui');
+const DIST = path.join(CWD, 'dist');
+const INDEX_TEMPLATE = path.join(UI, 'index.html');
+const ADMIN_TEMPLATE = path.join(UI, 'admin.html');
+const PERF_TEMPLATE = path.join(UI, 'perf.html');
+const LOGVIEWER_TEMPLATE = path.join(UI, 'logviewer.html');
+const FAILUREVIEWER_TEMPLATE = path.join(UI, 'failureviewer.html');
+const USERGUIDE_TEMPLATE = path.join(UI, 'userguide.html');
+
+const HTML_MINIFY_OPTIONS = {
+    useShortDoctype: true,
+    keepClosingSlash: true,
+    collapseWhitespace: true,
+    preserveLineBreaks: true
+};
+
+module.exports = neutrino => {
+    reactPreset(neutrino);
+
+    // Change the ouput path from build/ to dist/:
+    neutrino.config.output.path(DIST);
+
+    neutrino.config.module
+        .rule('lint')
+        .pre()
+        .test(/\.jsx?/)
+        .include(UI)
+        .loader('eslint', require.resolve('eslint-loader'), {
+            configFile: path.join(CWD, '.eslintrc')
+        });
+
+    // Neutrino looks for the entry at src/index.js by default; Delete this and add the index in ui/:
+    neutrino.config
+        .entry('index')
+        .delete(path.join(SRC, 'index.js'))
+        .add(path.join(UI, 'entry-index.js'))
+        .end();
+    // Add several other treeherder entry points:
+    neutrino.config
+        .entry('admin')
+        .add(path.join(UI, 'entry-admin.js'))
+        .end();
+    neutrino.config
+        .entry('perf')
+        .add(path.join(UI, 'entry-perf.js'))
+        .end();
+    neutrino.config
+        .entry('logviewer')
+        .add(path.join(UI, 'entry-logviewer.js'))
+        .end();
+    neutrino.config
+        .entry('failureviewer')
+        .add(path.join(UI, 'entry-failureviewer.js'))
+        .end();
+    neutrino.config
+        .entry('userguide')
+        .add(path.join(UI, 'entry-userguide.js'))
+        .end();
+
+    // Include commonly used files in a vendor chunk:
```

Comment:
That's the kind of behavior that's going on here, but I initially configured it to just bundle anything used twice or more into the vendor chunk, since that seemed to produce the overall smallest bundle. Agreed it would be better to base the division on whether the code comes from vendor libs, as you describe - that's easy to set up - I'll do it today.